pkg/chunk/chunk.go

```diff
@@ -333,27 +327,3 @@ func (c *Chunk) Samples(from, through model.Time) ([]model.SamplePair, error) {
 	interval := metric.Interval{OldestInclusive: from, NewestInclusive: through}
 	return prom_chunk.RangeValues(it, interval)
 }
-
-// Slice builds a new smaller chunk with data only from given time range (inclusive)
-func (c *Chunk) Slice(from, through model.Time) (*Chunk, error) {
-	// there should be atleast some overlap between chunk interval and slice interval
-	if from > c.Through || through < c.From {
-		return nil, ErrSliceOutOfRange
-	}
-
-	pc, err := c.Data.Rebound(from, through)
-	if err != nil {
-		return nil, err
-	}
-
-	nc := NewChunk(c.UserID, c.Fingerprint, c.Metric, pc, from, through)
-	return &nc, nil
-}
-
-func intervalsOverlap(interval1, interval2 model.Interval) bool {
-	if interval1.Start > interval2.End || interval2.Start > interval1.End {
-		return false
-	}
-
-	return true
-}
```

Comment:
pkg/chunk/chunk.go mentions dynamodb a few times. I think more functions can be removed from here, but I am not sure.

Reply:
Yeah, this is pretty messy. I'll rip out much of this in an upcoming commit on this PR, but I'll stop before a wholesale refactor to avoid this PR getting even larger.